services/blockchainService.ts

```diff
@@ -27,12 +18,23 @@ export interface BlockInfo extends BlockchainInfo {
   timestamp: number
 }
 
+export interface TransfersResponse {
```

Comment:
Why call those `Transfer` where everywhere else this is called `Transaction`?

Ofc `Transaction` already exists as a `Prisma`  generated type, but this shouldn't be `Transaction` either since it wouldn't be clear that it refers to the `blockchainService` expected inputs or outputs, you could just name the single one `TransactionResponse` and this one `TransactionsResponse`

Reply:
In general, I just think it's a bad practice to take two words that are kind of synonyms and make such a convention that the one means X, the other one means Y, since then the name would not help anybody who reads the code and tries to understand what the structures are.

Reply:
Maybe we can find a better name, however a blockchain transaction is a concept where we have possibly many inputs and many outputs, that is not the data structure we have here, here we have just how much money is going in or out of one address and when, similar to one `Transfer`, came to mind. Maybe `Credit`? Anyway, I feel that calling this simple data structure a `Transaction` is conceptually wrong.

Note that this is an interpretation of the blockchain data done after some data crunching and transforming,  we can (and did) generate a new structure and data out of it.

Reply:
Yes, you have a good point. Then the DB table should also be called `Transfer` instead of `Transaction`, right?
Actually, now I am asking why don't we use the prisma generated type here? In the end the idea is put those in that table anyways. Then what you called `txid` would be `hash` and `receivedAmount` would be just `amount`.

I can see the difference now and agree that `Transfer` is indeed a better name (it even helps separate `prisma.transaction` from `prisma.$transaction`), as `txid` and `receivedAmount` also are better, clearer than `hash` and `amount`. So I actually think it would be better to make a refactor in the future, change the name of the prisma table & fields.

But a more important question for now, regarding using the prisma generated types here (now or after the refactor): shouldn't we add an `addressId` and a `confirmed` fields to this type? Then it would be fully compatible with the prisma types.

Reply:
can you add the other two field tho?

Reply:
Done. I added the Address instead of number addressId since in PR 411-6 I already need an Address in there.

Reply:
I actually still can't see the answer for:
> Actually, now I am asking why don't we use the prisma generated type here?

The reasons I see is that the name `Transfer` is better as are the field names, but since all the codebase already uses the prisma `Transaction` table, it doesn't seem worth to me to create a virtually identical type. In some future refactor we can and should change the name of the Transaction table & its fields, but for now you wouldn't even have to define an extra interface here; simply use the one already provided by prisma